File: app/src/main/java/me/imzack/app/ender/injector/component/HomeComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.HomePresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.HomeActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = HomePresenterModule.class)
public interface HomeComponent {

    void inject(HomeActivity homeActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/HomeComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.HomePresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.HomeActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(HomePresenterModule::class))
interface HomeComponent {

    fun inject(homeActivity: HomeActivity)
}
